rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,339 @@
+- File Slice determination logic for log files changed. In 0.x, we have base instant time in log files and its straight forward. 
+In 1.x, we find completion time for a log file and find the base instant time (parsed from base files) for a given HoodieFileGroup, 
+- which has the highest value lesser than the completion time of the log file) of interest. 
+- Log file ordering within a file slice. (in 0.x, we use base instant time ➝ log file versions ➝ write token) to order diff log files. in 1.x, we will be using completion time to order).
+- Rollbacks in 0.x appends a new rollback block (new log file). While in 1.x, rollback will remove the partially failed log files. 
+
+### Log format changes:
+- We have added a new header type, IS_PARTIAL in 1.x.
+
+## Changes to be ported over to 0.16.x to support reading 1.x tables
+
+### What will be supported
+- For features introduced in 0.x, and tables written in 1.x, 0.16.0 reader should be able to provide consistent reads w/o any breakage. 
+
+### What will not be supported
+- A 0.16 writer cannot write to a table that has been upgraded-to/created using 1.x without downgrading to 0.16. Might be obvious, but calling it out nevertheless.
+- For new features introduced in 1.x, we may or may not have full support with 0.16.x reader. 
+
+| 1.x Features written by 1.x writer        | 1.x reader   | 0.16.x Reader | 
+|----------------|-------------------------------------------|-------|
+| Deletion vector | supported | Falls back to key based merges giving up on perf optimization
+| Partial merges/updates  | supported | Fails with clear error message stating that partial merging |
+| Functional indexes | supported | Not supported. Perf optimization may not kick in|
+| Secondary indexes | supported | Not supported. Perf optimization may not kick in|
+| NBCC OR Completion time based log file ordering in a file slice | supported | Not supported. Will be using log file version and write token based ordering.|  
+
+
+### Timeline
+- Commit instants w/ completion time should be readable. HoodieInstant parsing logic to parse completion time should be ported over.
+- Commit metadata in avro instead of json should be readable.
+   - More details on this under Implementation section.
+- Pending Clustering commits using “cluster” action should be readable in 0.16.0 reader.
+- HoodieDefaultTimeline should be able to support both 0.x timeline and 1.x timeline.
+   - More details on this under Implementation section.
+- Should we port LSM reader as well to 0.16.x? Our goal here is to support snapshot, time travel and incremental queries for 1.x tables. Strictly speaking we can only do all these 3 queries in uncleaned 
+instants i.e. over instant ranges where cleaner has not been executed. So, if we guarantee that 1.x active timeline will definitely contain all the uncleaned instants, we could get away by not even porting over 
```

Comment:
I think most of the reader logic is based on the active timeline instants, and the 1.x archival strategy does not really change comparing to 0.x, but we may still need to read the LSM archived timeline if there are 1.x written log files in the table while the 0.x reader want to read them(for the completion time).

Reply:
we don't need completion time in 0.16.0 reader if we do not need to support NBCC. as per feedback, I am going to add a section in this RFC on bridge release design if we were to support NBCC as well. To get that working, we will be needing completion time query view.